File: app/src/main/java/com/frogobox/board/ui/SplashActivity.java
package com.frogobox.board.ui;

import android.content.Intent;
import android.os.Bundle;

import com.frogobox.board.base.BaseActivity;
import com.frogobox.board.util.FirstLaunchManager;


public class SplashActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Intent mainIntent;

        FirstLaunchManager firstStartPref = new FirstLaunchManager(this);

        if (firstStartPref.isFirstTimeLaunch()) {
            firstStartPref.initFirstTimeLaunch();
            mainIntent = new Intent(this, TutorialActivity.class);
        } else {
            mainIntent = new Intent(this, MainActivity.class);
            mainIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
        }

        startActivity(mainIntent);
        finish();
    }

}

File: app/src/main/java/com/frogobox/board/mvvm/main/SplashActivity.kt
package com.frogobox.board.mvvm.main

import android.content.Intent
import android.os.Bundle
import com.frogobox.board.core.BaseActivity
import com.frogobox.board.mvvm.tutorial.TutorialActivity
import com.frogobox.board.mvvm.tutorial.TutorialController

class SplashActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val mainIntent: Intent
        val firstStartPref = TutorialController(this)

        if (firstStartPref.isFirstTimeLaunch) {
            firstStartPref.initFirstTimeLaunch()
            mainIntent = Intent(this, TutorialActivity::class.java)
        } else {
            mainIntent = Intent(this, MainActivity::class.java)
            mainIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
        }

        startActivity(mainIntent)
        finish()
    }
}